chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/component/ChaosMonkeyScheduler.java

```diff
@@ -1,49 +1,83 @@
 package de.codecentric.spring.boot.chaos.monkey.component;
 
+import de.codecentric.spring.boot.chaos.monkey.assaults.ChaosMonkeyRuntimeAssault;
 import de.codecentric.spring.boot.chaos.monkey.configuration.AssaultProperties;
+import java.util.*;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.lang.Nullable;
 import org.springframework.scheduling.config.CronTask;
 import org.springframework.scheduling.config.ScheduledTask;
 import org.springframework.scheduling.config.ScheduledTaskRegistrar;
 
+/**
+ * @author Maxime Bouchenoire
+ * @author Lukas Morawietz
+ */
 public class ChaosMonkeyScheduler {
 
   private static final Logger Logger = LoggerFactory.getLogger(ChaosMonkeyScheduler.class);
 
-  @Nullable private final ScheduledTaskRegistrar scheduler;
+  private final ScheduledTaskRegistrar scheduler;
 
   private final AssaultProperties config;
+  private final List<ChaosMonkeyRuntimeAssault> assaults;
 
-  private final ChaosMonkeyRuntimeScope runtimeScope;
-
-  @Nullable private ScheduledTask currentTask;
+  private final Map<ChaosMonkeyRuntimeAssault, ScheduledTask> currentTasks = new HashMap<>();
 
   public ChaosMonkeyScheduler(
       ScheduledTaskRegistrar scheduler,
       AssaultProperties config,
-      ChaosMonkeyRuntimeScope runtimeScope) {
+      List<ChaosMonkeyRuntimeAssault> assaults) {
     this.scheduler = scheduler;
     this.config = config;
-    this.runtimeScope = runtimeScope;
+    this.assaults = assaults;
 
     reloadConfig();
   }
 
   public void reloadConfig() {
-    String cronExpression = config.getRuntimeAssaultCronExpression();
-    boolean active = !"OFF".equals(cronExpression);
-
-    if (currentTask != null) {
-      Logger.info("Cancelling previous task");
-      currentTask.cancel();
-      currentTask = null;
+    Map<ChaosMonkeyRuntimeAssault, String> cronExpressions = new LinkedHashMap<>();
+    for (ChaosMonkeyRuntimeAssault assault : assaults) {
+      cronExpressions.put(assault, assault.getCronExpression(config));
     }
-
-    if (active) {
-      CronTask task = new CronTask(runtimeScope::callChaosMonkey, cronExpression);
-      currentTask = scheduler.scheduleCronTask(task);
+    if (!currentTasks.isEmpty()) {
+      for (Iterator<Map.Entry<ChaosMonkeyRuntimeAssault, String>> iterator =
+              cronExpressions.entrySet().iterator();
+          iterator.hasNext(); ) {
+        Map.Entry<ChaosMonkeyRuntimeAssault, String> entry = iterator.next();
+        ScheduledTask task = currentTasks.get(entry.getKey());
+        if (task != null) {
+          if (Objects.equals(((CronTask) task.getTask()).getExpression(), entry.getValue())) {
```

Comment:
Dont like this wild cast here (allthough we should be safe here as we only put CronTasks :D), but not really sure if we can improve it.
When we would do an instance check it will be typesafe but might lead to bugs.